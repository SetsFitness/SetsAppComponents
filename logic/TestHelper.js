import ably from "../redux/reducers/ablyReducer";
import auth from "../redux/reducers/authReducer";
import cache from "../redux/reducers/cacheReducer";
import info from "../redux/reducers/infoReducer";
import message from "../redux/reducers/messageReducer";
import search from "../redux/reducers/searchReducer";
import user from "../redux/reducers/userReducer";

/**
 * Class used for testing certain parts of the code that may try to connect to AWS in order to complete. Run this when
 * we are automatically running a test.
 */
export default class TestHelper {
    static ifTesting = false;
    static setTest() { TestHelper.ifTesting = true; }
    static unsetTest() { TestHelper.ifTesting = false; }
}

/**
 * Gets the entire redux store components given for initializing a mock store. Allows you to only initialize what you
 * need to for the test.
 *
 * TODO Edit when we use more reducers.
 *
 * @param {[string]} storeTypes The strings for the types of the redux store to initialize.
 * @return {{ably: *, auth: *, cache: *, info: *, message: *, user: *, search: *}}
 */
export const getInitalReduxStore = (storeTypes) => {
    return {
        ably: !storeTypes || storeTypes.includes("ably") ? ably(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        auth: !storeTypes || storeTypes.includes("auth") ? auth(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        cache: !storeTypes || storeTypes.includes("cache") ? cache(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        info: !storeTypes || storeTypes.includes("info") ? info(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        message: !storeTypes || storeTypes.includes("message") ? message(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        user: !storeTypes || storeTypes.includes("user") ? user(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
        search: !storeTypes || storeTypes.includes("search") ? search(undefined, {type: "__NOT_A_REAL_ACTION__"}) : null,
    }
};