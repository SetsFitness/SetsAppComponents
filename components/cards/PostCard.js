import React, {useState} from 'react';
import {Button, Card, Dimmer, Grid} from 'semantic-ui-react';
import {connect} from "react-redux";
import PostDescriptionModal from '../modals/PostDescriptionModal';
import {Player} from "video-react";
import ItemType from "../../logic/ItemType";
import ChallengeDetailCard, {ChallengeDetailCardInfo} from "./post_detail_cards/ChallengeDetailCard";
import PostDetailCard from "./post_detail_cards/PostDetailCard";
import ClientDetailCard from "./post_detail_cards/ClientDetailCard";
import TrainerDetailCard from "./post_detail_cards/TrainerDetailCard";
import {convertFromISO} from "../../logic/TimeHelper";
import ClientModal from "../modals/ClientModal";
import TrainerModal from "../modals/TrainerModal";
import {GroupDetailCardInfo} from "./post_detail_cards/GroupDetailCard";
import {EventDetailCardInfo} from "./post_detail_cards/EventDetailCard";
import Spinner from "../props/Spinner";
import {getAttributeFromObject, getObjectAttribute, getObjectFromCache} from "../../logic/CacheRetrievalHelper";
import StyledProfileImage from "../props/StyledProfileImage";
import type Post from "../../types/Post";

export const PostCardInfo = {
  fetchList: ["id", "time_created", "by", "item_type", "postType", "about", "access", "description", "videoPaths", "picturePaths"],
  byInfo: {
    clientFetchList: ["id", "name", "friends", "challengesWon", "scheduledEvents", "profileImagePath"],
    trainerFetchList: ["id", "name", "gender", "birthday", "profileImagePath", "profileImagePaths"],
  },
  aboutInfo: {
    clientFetchList: ["id", "name", "friends", "challengesWon", "scheduledEvents", "profileImagePath"],
    trainerFetchList: ["id", "name", "gender", "birthday", "profileImagePath", "profileImagePaths"],
    eventFetchList: EventDetailCardInfo.fetchList,
    challengeFetchList: ChallengeDetailCardInfo.fetchList,
    postFetchList: ["id", "time_created", "by", "item_type", "postType", "about", "access", "description", "videoPaths", "picturePaths"],
    groupFetchList: GroupDetailCardInfo.fetchList,
  },
  ifSubscribe: false,
};

type Props = {
  post: Post
  // by: {
  //     id: string,
  //     item_type: string,
  //     name: string,
  //     profileImagePath: string,
  //     profileImage: string,
  // },
  // about?: {
  //     id: string,
  //     item_type: string,
  //     title: string,
  //     owner: string
  // }
};

/**
 * Displays the profile image.
 *
 * @param {string} profileImage The url string for the profile image.
 * @returns {*} Displays the JSX of the profile image given.
 */
export const profilePicture = (profileImage) => {
  if (profileImage) {
    return (
      <StyledProfileImage profileImage={profileImage} type="Small"/>
      //<div align="center" className="ui u-avatar tiny" style={{backgroundImage: `url(${profileImage})`, width: '50px', height: '50px'}}/>
    );
  } else {
    return (
      <Dimmer inverted>
        <Spinner className="spinner1"/>
      </Dimmer>
    );
  }
};

/**
 * Displays the video or image media in the post.
 *
 * @param {[]} pictures A list of all the pictures in the post.
 * @param {[]} videos A list of all the videos in the post.
 * @returns {*} The React JSX used to display the video or image in a post.
 */
export const getPostMedia = (pictures, videos) => {
  // TODO FIX THIS UP SO THAT IT CAN HANDLE A LOT MORE POSTS
  if (videos && videos.length > 0) {
    return (
      <Player inline={true}>
        <source src={videos[0]} type="video/mp4"/>
      </Player>
    );
  }
  return null;
};

/**
 * Displays the correct detail card based on the type of post that it is.
 *
 * @param {string} postType The type of post that is being displayed.
 * @param {string} about The id of the content that the post is about.
 * @returns {*}
 */
export const getCorrectDetailCard = (postType, about) => {
  if (postType && postType.length) {
    //alert("Item Type: " + itemType);
    if (postType.substr(0, 3) === "new") {
      // TODO This indicates that this is for a newly created Item
      postType = ItemType[postType.substring(3, postType.length)];
    }
    //alert(itemType);
    if (postType) {
      //alert("Post Type: " + postType);
      // TODO Switch the post types
      if (postType === "Client") {
        //alert("Client Share Post Spotted!");
        // if(!this.state.postMessageSet) {
        //     this.setState({postMessage: "shared a user profile", postMessageSet: true});
        // }
        return (<ClientDetailCard client={about}/>);
      } else if (postType === "Trainer") {
        // if(!this.state.postMessageSet) {
        //     this.setState({postMessage: "shared a trainer profile", postMessageSet: true});
        // }
        return (<TrainerDetailCard trainer={about}/>);
      } else if (postType === "Gym") {
        //return (<GymDetailCard displayMedia = {this.getDisplayMedia}/>);
      } else if (postType === "Workout") {
        //return (<WorkoutDetailCard displayMedia = {this.getDisplayMedia}/>);
      } else if (postType === "Review") {
        //return (<ReviewDetailCard displayMedia = {this.getDisplayMedia}/>);
      } else if (postType === "Event") {
        //return (<EventDetailCard displayMedia = {this.getDisplayMedia}/>);
      } else if (postType === "Challenge") {
        // if(!this.state.postMessageSet) {
        //     this.setState({postMessage: "shared a challenge", postMessageSet: true});
        // }
        return (<ChallengeDetailCard challenge={about}/>);
      } else if (postType === "Invite") {
        //return (<InviteDetailCard displayMedia = {this.getDisplayMedia}/>);
      } else if (postType === "Post") {
        // if(!this.state.postMessageSet) {
        //     this.setState({postMessage: "posted", postMessageSet: true});
        // }
        return (<PostDetailCard postID={about.id}/>);
      }
    }
  }
  return (<div/>);
};

/**
 * The modal of the user that the post is authored by.
 *
 * @param {string} id The id of the user that created the modal.
 * @param {string} itemType
 * @param {function()} byModalOpen Opens the modal relating to the user who authored the post.
 * @param {function(boolean)} setByModalOpen Sets the by modal open attribute to true.
 * @returns {*} The React JSX user to display the proper client modal state.
 */
export const byModal = (id, itemType, byModalOpen, setByModalOpen) => {
  if (itemType === "Client") {
    return (
      <ClientModal open={byModalOpen} onClose={() => setByModalOpen(false)} clientID={id}/>
    );
  } else if (itemType === "Trainer") {
    return (
      <TrainerModal open={byModalOpen} onClose={() => setByModalOpen(false)} trainerID={id}/>
    );
  }
  return null;
};

/**
 * This is the generic view for how a post shows up in any feeds or lists.
 * It is used as a modal trigger in the feed.
 *
 * @param {Props} props The given props to the component.
 * @returns {*} The React JSX used to display the component.
 * @constructor
 */
const PostCard = (props: Props) => {
  const [postModalOpen, setPostModalOpen] = useState(false);
  const [byModalOpen, setByModalOpen] = useState(false);

  const getPostAttribute = (attributeName) => {
    return getAttributeFromObject(props.post, attributeName);
  };
  const getByAttribute = (attributeName) => {
    return getObjectAttribute(getPostAttribute("by"), attributeName, props.cache);
  };
  // const getAboutAttribute = (attributeName) => {
  //     return getAttributeFromObject(props.about, attributeName, props.cache);
  // };
  // const getBy = () => {
  //     return getObjectFromCache(props.by, props.cache);
  // };
  const getAbout = () => {
    return getObjectFromCache(getPostAttribute("about"), props.cache);
  };

  if (!props.post || !getByAttribute("name")) {
    return (
      <Card color='purple' fluid raised>
        <Spinner loading={true}/>
      </Card>
    );
  } else {
    return (
      <Card color='purple' fluid raised>
        {/*this.getPostAttribute("about")*/}
        {/*{openOnce()}*/}
        <Card.Content>
          <div align='center'>
            {getCorrectDetailCard(getPostAttribute("postType"), getAbout())}
          </div>
          {/*this.getDisplayMedia()*/}
        </Card.Content>
        <Card.Content extra onClick={() => postModalOpen || setPostModalOpen(true)}>
          {/*<Card.Meta textAlign = 'center'>{this.getPostAttribute("description")}</Card.Meta>*/}
          <div align="center">
            {convertFromISO(getPostAttribute("time_created")).substr(5, 12)}
          </div>
          <PostDescriptionModal open={postModalOpen} onClose={() => setPostModalOpen(false)}
                                postID={props.post.id}/>
        </Card.Content>
      </Card>
    );
  }
};

const mapStateToProps = state => ({
  cache: state.cache
});

// const connector: Connector<Props, any> = connect(mapStateToProps);
// export default connector(PostCard);
export default connect(mapStateToProps)(PostCard);
