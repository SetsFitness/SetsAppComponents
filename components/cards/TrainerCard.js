import React, {useState} from 'react';
import {Card, Dimmer, Icon, Button} from 'semantic-ui-react';
import TrainerPortalModal from '../modals/TrainerModal';
import {getAttributeFromObject} from "../../logic/CacheRetrievalHelper";
import Spinner from "../props/Spinner";
import StyledProfileImage from "../props/StyledProfileImage";
import type Trainer from "../../types/Trainer";

export const TrainerCardInfo = {
  fetchList: ["id", "name", "profileImagePath"],
  ifSubscribe: false
};


type Props = {
  rank?: number,
  trainer: Trainer
}

/**
 * This is the generic view for how a trainer shows up in any feeds or lists.
 * It is used as a modal trigger in the feed.
 *
 * @param {Props} props The given props to the component.
 * @returns {*} The React JSX used to display the component.
 * @constructor
 */
const TrainerCard = (props: Props) => {
  const [modalOpen, setModalOpen] = useState(false);
  const getTrainerAttribute = (attributeName) => {
    return getAttributeFromObject(props.trainer, attributeName);
  };

  if (!props.trainer) {
    return (
      <Dimmer inverted>
        <Spinner/>
      </Dimmer>
    );
  }
  return (
    <Card fluid raised centered onClick={() => modalOpen || setModalOpen(true)}>
      <div className="u-container">
        <StyledProfileImage profileImage={getTrainerAttribute("profileImage")} type="LargeCenter"/>
      </div>
      <Card.Content textAlign='center'>
        <Card.Header>
          {getTrainerAttribute("name")}
        </Card.Header>
        <Card.Description>
          {/*this.getTrainerAttribute("title")*/} Personal Trainer
        </Card.Description>
      </Card.Content>
      <Card.Content extra textAlign='center'>
        <Button.Group fluid>
          <Button>
            <Icon name='globe'/> Boston, MA</Button>
          <Button primary>
            Vastus Certified</Button>
        </Button.Group>
      </Card.Content>
      <TrainerPortalModal open={modalOpen} onClose={() => setModalOpen(false)} trainerID={getTrainerAttribute("id")}/>
    </Card>
  )
};

export default TrainerCard;

