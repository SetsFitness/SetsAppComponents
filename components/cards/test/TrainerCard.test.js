import React from 'react';
import {Provider} from "react-redux";
import {shallow, mount} from 'enzyme';
import TrainerCard from "../TrainerCard";
import "../../../testing/SetTesting";
import {store} from "../../../testing/TestHelper";
import {funCompWrapper} from "../../../testing/TestHelper";
import {profilePicture} from "../PostCard";
import {expect} from "chai";

const reduxState = {
  cache: {
    trainers: {
      TR0001: {
        id: "TR0001",
        profileImage: "thisIsAFakePath/Path",
        name: "Sleeny"
      }
    },
    challenges: {
      CH0001: {
        id: "CH0001",
        by: "CL0001",
      }
    }
  },
};

const trainer1 = {
  id: "TR0001",
  profileImage: "thisIsAFakePath/Path",
  name: "Sleeny"
};

it('renders without crashing', () => {
  const component = shallow(<TrainerCard store={store()} rank={1} trainer={trainer1}/>);
  global.expect(component).toMatchSnapshot();
});

describe("Post Card", () => {
  // States ~
  describe("States", () => {
    // Here you check the state of the component, like the shown text and sub-components for the component.
    // Make sure you go into all the conditions of the component so that you test every possible state of the
    // component.
    // EXAMPLE: it("Shows the spinner when no post is passed in");
    it('trainer card without trainer prop displays spinner', () => {
      const component = shallow(<TrainerCard store={store()} rank={1} trainer={null}/>);

      expect(component.find("Spinner")).to.have.lengthOf(1);
    });

    it('trainer card with trainer prop displays card', () => {
      const component = shallow(<TrainerCard store={store()} rank={1} trainer={trainer1}/>);

      expect(component.find("Card")).to.have.lengthOf(1);
    });

    it('trainer card with trainer prop displays trainer name in header', () => {
      const component = shallow(<TrainerCard store={store(reduxState)} rank={1} trainer={trainer1}/>);

      console.log(component.find("CardHeader").render().text());

      expect(component.find("CardHeader").render().text()).equals("Sleeny");
    });
  });
  // Events ~
  describe("Events", () => {
    // Here you use .simultate('click') to check the change in the component.
    // Only in special circumstances should you use jest.fn() and that's if you're directly passing in a function
    // into the component's props and you have access to it.

    //TODO: Test that pressing the card sets descriptionModalOpen to true
    it('trainer card with trainer prop displays trainer name in header', () => {
      const component = shallow(<TrainerCard store={store(reduxState)} rank={1} trainer={trainer1}/>);

      expect(component.find("CardHeader").render().text()).equals("Sleeny");
    });
  });
});
