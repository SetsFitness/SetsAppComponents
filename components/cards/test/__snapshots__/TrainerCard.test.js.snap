// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrainerCard
    rank={1}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    trainer={
      Object {
        "TR0001": Object {
          "id": "TR0001",
          "name": "Sleeny",
          "profileImage": "thisIsAFakePath/Path",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "centered": true,
      "children": Array [
        <div
          className="u-container"
        >
          <StyledProfileImage
            type="LargeCenter"
          />
        </div>,
        <CardContent
          textAlign="center"
        >
          <CardHeader />
          <CardDescription>
             Personal Trainer
          </CardDescription>
        </CardContent>,
        <CardContent
          extra={true}
          textAlign="center"
        >
          <ButtonGroup
            fluid={true}
          >
            <Button
              as="button"
              role="button"
            >
              <Icon
                as="i"
                name="globe"
              />
               Boston, MA
            </Button>
            <Button
              as="button"
              primary={true}
              role="button"
            >
              Vastus Certified
            </Button>
          </ButtonGroup>
        </CardContent>,
        <Connect(TrainerModal)
          onClose={[Function]}
          open={false}
        />,
      ],
      "fluid": true,
      "onClick": [Function],
      "raised": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <StyledProfileImage
            type="LargeCenter"
          />,
          "className": "u-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "profileImage": undefined,
            "type": "LargeCenter",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardHeader />,
            <CardDescription>
               Personal Trainer
            </CardDescription>,
          ],
          "textAlign": "center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": " Personal Trainer",
            },
            "ref": null,
            "rendered": " Personal Trainer",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ButtonGroup
            fluid={true}
          >
            <Button
              as="button"
              role="button"
            >
              <Icon
                as="i"
                name="globe"
              />
               Boston, MA
            </Button>
            <Button
              as="button"
              primary={true}
              role="button"
            >
              Vastus Certified
            </Button>
          </ButtonGroup>,
          "extra": true,
          "textAlign": "center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Button
                as="button"
                role="button"
              >
                <Icon
                  as="i"
                  name="globe"
                />
                 Boston, MA
              </Button>,
              <Button
                as="button"
                primary={true}
                role="button"
              >
                Vastus Certified
              </Button>,
            ],
            "fluid": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "children": Array [
                  <Icon
                    as="i"
                    name="globe"
                  />,
                  " Boston, MA",
                ],
                "role": "button",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "i",
                    "name": "globe",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " Boston, MA",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "children": "Vastus Certified",
                "primary": true,
                "role": "button",
              },
              "ref": null,
              "rendered": "Vastus Certified",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClose": [Function],
          "open": false,
          "trainerID": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "centered": true,
        "children": Array [
          <div
            className="u-container"
          >
            <StyledProfileImage
              type="LargeCenter"
            />
          </div>,
          <CardContent
            textAlign="center"
          >
            <CardHeader />
            <CardDescription>
               Personal Trainer
            </CardDescription>
          </CardContent>,
          <CardContent
            extra={true}
            textAlign="center"
          >
            <ButtonGroup
              fluid={true}
            >
              <Button
                as="button"
                role="button"
              >
                <Icon
                  as="i"
                  name="globe"
                />
                 Boston, MA
              </Button>
              <Button
                as="button"
                primary={true}
                role="button"
              >
                Vastus Certified
              </Button>
            </ButtonGroup>
          </CardContent>,
          <Connect(TrainerModal)
            onClose={[Function]}
            open={false}
          />,
        ],
        "fluid": true,
        "onClick": [Function],
        "raised": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <StyledProfileImage
              type="LargeCenter"
            />,
            "className": "u-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "profileImage": undefined,
              "type": "LargeCenter",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <CardHeader />,
              <CardDescription>
                 Personal Trainer
              </CardDescription>,
            ],
            "textAlign": "center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": " Personal Trainer",
              },
              "ref": null,
              "rendered": " Personal Trainer",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ButtonGroup
              fluid={true}
            >
              <Button
                as="button"
                role="button"
              >
                <Icon
                  as="i"
                  name="globe"
                />
                 Boston, MA
              </Button>
              <Button
                as="button"
                primary={true}
                role="button"
              >
                Vastus Certified
              </Button>
            </ButtonGroup>,
            "extra": true,
            "textAlign": "center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Button
                  as="button"
                  role="button"
                >
                  <Icon
                    as="i"
                    name="globe"
                  />
                   Boston, MA
                </Button>,
                <Button
                  as="button"
                  primary={true}
                  role="button"
                >
                  Vastus Certified
                </Button>,
              ],
              "fluid": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "button",
                  "children": Array [
                    <Icon
                      as="i"
                      name="globe"
                    />,
                    " Boston, MA",
                  ],
                  "role": "button",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": "i",
                      "name": "globe",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " Boston, MA",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "button",
                  "children": "Vastus Certified",
                  "primary": true,
                  "role": "button",
                },
                "ref": null,
                "rendered": "Vastus Certified",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClose": [Function],
            "open": false,
            "trainerID": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
